megamek/src/megamek/common/weapons/ACAPHandler.java

```diff
@@ -96,41 +97,20 @@ protected void handleEntityDamage(Entity entityTarget,
         }
 
 
-        // A building may be damaged, even if the squad is not.
-        if (bldgAbsorbs > 0) {
-            int toBldg = Math.min(bldgAbsorbs, nDamage);
-            nDamage -= toBldg;
-            Report.addNewline(vPhaseReport);
-            Vector<Report> buildingReport = server.damageBuilding(bldg, toBldg,
-                    entityTarget.getPosition());
-            for (Report report : buildingReport) {
-                report.subject = subjectId;
-            }
-            vPhaseReport.addAll(buildingReport);
-        // Units on same level, report building absorbs no damage
-        } else if (bldgAbsorbs == Integer.MIN_VALUE) {
-            Report.addNewline(vPhaseReport);
-            r = new Report(9976);
-            r.subject = ae.getId();
-            r.indent(2);
-            vPhaseReport.add(r);
-        // Cases where absorbed damage doesn't reduce incoming damage
-        } else if (bldgAbsorbs < 0) {
-            int toBldg = -bldgAbsorbs;
-            Report.addNewline(vPhaseReport);
-            Vector<Report> buildingReport = server.damageBuilding(bldg, toBldg,
-                    entityTarget.getPosition());
-            for (Report report : buildingReport) {
-                report.subject = subjectId;
-            }
-            vPhaseReport.addAll(buildingReport);
-        }
+        // if the target was in partial cover, then we already handled
+        // damage absorption by the partial cover, if it would have happened
+        IHex targetHex = game.getBoard().getHex(target.getPosition());
```

Comment:
Does this damage handler (or subsequent handlers) apply to offboard units as well? Said another way: could `targetHex` be null here?

Reply:
No, this is a handler for autocannons loaded with armor piercing ammo.

Reply:
Introduced null checking for hex out of an abundance of caution, and needed extra logic in there anyway.